### ISSUE TITLE
Current compiling process relies on SVN for setting Kratos version

### ISSUE THREAD

**Issue by marandra** [KratosMultiphysics/Kratos#9]
The current way of setting the version number relies on a .svn directory, so a fresh cloned git repository in a system with subversion detected will fail to compile.

I don't think we can maintain the versioning system already used with svn (incremental commit number), we should probably develop a process that encodes tags and commit hashes in the source.

**Comment by pooyan-dadvand** [comment]
Definitely we should change that. It was very SVN way of release. Do you know any other project who doing it well?

**Comment by ddemidov** [comment]
I am using git revision as a version descriptor in one of my projects. CMake parses the output of `git rev-parse --short HEAD` [here](https://bitbucket.org/d_demidov/microseism/src/b9f56eb93d381e4e07559838b78f80480e83ae72/CMakeLists.txt?at=master&fileviewer=file-view-default#CMakeLists.txt-18:22), and [pastes](https://bitbucket.org/d_demidov/microseism/src/b9f56eb93d381e4e07559838b78f80480e83ae72/CMakeLists.txt?at=master&fileviewer=file-view-default#CMakeLists.txt-105:108) the result to a [header file](https://bitbucket.org/d_demidov/microseism/src/b9f56eb93d381e4e07559838b78f80480e83ae72/config.h.cmake?at=master&fileviewer=file-view-default#config.h.cmake-34).

**Comment by pooyan-dadvand** [comment]
Thank you Denis! 
Is there anyway to retrieve the code using this hash number? (to reproduce an error of users having this number in log file)

**Comment by ddemidov** [comment]
Yes, `git rev-parse --short HEAD` should give you a shortened SHA1, which you can use with `git checkout <rev>`. Here is an example:
```
$ git rev-parse --short HEAD
0d4edd2
$ git checkout 0d4edd2
Note: checking out '0d4edd2'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 0d4edd2...
```

**Comment by pooyan-dadvand** [comment]
Perfect!! Thanks!
